### PR TITLE
[Feature]Add evm chain methods

### PR DESCRIPTION
**Pull request by akira02**
## Summary
Add `wallet_addEthereumChain` and `wallet_switchEthereumChain` rpc method.
Add `loadSwitchableNetwork` function to sdk, for one-time load all switchable chain id. It's because `wallet_addEthereumChain` only allow one chain at a time.

Can discuss about the method name.

## Related Links

**Asana**:  https://app.asana.com/0/1201812548509877/1204162132560893/f

**Mockup**:  N/A

## Checklist

- [x] Pasted Asana or Mockup link.
- [x] Tagged labels.

## NOTE
Won't check api when new the sdk class because of async in constructor issue. Will change sdk struc later.

### CODE REVIEW

File: src/providers/ethereum.ts [hunk 1]
```diff
@@ -16,34 +20,49 @@ import {
...
+    network_type: string
+    wallet_web_url: string
+    rpc_url: string
+  }
+}
+
+function parseChainId(chainId: string | number): number {
+  if (typeof chainId === 'number') {
+    return chainId;
+  } else if (chainId.includes('0x')) {
```
Comment (moojing) [inline src/providers/ethereum.ts:44]: Maybe `startWith`  is more precise?

File: src/providers/ethereum.ts [hunk 2]
```diff
@@ -206,6 +272,48 @@ export default class EthereumProvider extends BloctoProvider implements Ethereum
...
+        case 'wallet_switchEthereumChain':
+          if (!payload?.params?.[0]?.chainId) {
+            throw new Error('Invalid params');
+          }
+
+          if (!this.switchableNetwork[parseChainId(payload.params[0].chainId)]) {
+            const error: any = new Error(
+              'This chain has not been added to SDK. Please try wallet_addEthereumChain first.'
+            );
+            error.code = 4902;
```
Comment (moojing) [inline src/providers/ethereum.ts:299]: how does this 4902 be decided?

File: src/providers/types/ethereum.d.ts [hunk 3]
```diff
@@ -3,26 +3,36 @@ import { BaseConfig } from '../../constants';
...
+  rpc: string
+  server: string
+  accounts: Array<string>
+  request(args: EIP1193RequestPayload): Promise<any>
+  loadSwitchableNetwork(
+    networkList: {
+      chainId: string
+      rpcUrls?: string[]
+    }[]
+  ): Promise<null>
```
Comment (moojing) [inline src/providers/types/ethereum.d.ts:32]: do we need add `checkAndAddNetwork` method.

File: src/providers/ethereum.ts (hunk 2 above)
Comment (akira02) [reply src/providers/ethereum.ts:299]: Said by metamask but I also don't know why.
https://docs.metamask.io/guide/rpc-api.html#wallet-switchethereumchain

File: src/providers/ethereum.ts (hunk 1 above)
Comment (akira02) [reply src/providers/ethereum.ts:44]: Good point. Thanks.

File: src/providers/types/ethereum.d.ts (hunk 3 above)
Comment (akira02) [reply src/providers/types/ethereum.d.ts:32]: I think `loadSwitchableNetwork` is enough for use

File: src/providers/ethereum.ts (hunk 2 above)
Comment (akira02) [reply src/providers/ethereum.ts:300]: Comment added

**Review by moojing — APPROVED** [review]

**Pull request merged**